internal/pkg/deploy/cloudformation/cc_pipeline_integration_test.go

```diff
@@ -0,0 +1,182 @@
+					Accounts:     []*string{appStackInstance.Account},
+					Regions:      []*string{appStackInstance.Region},
+					RetainStacks: aws.Bool(false),
+					StackSetName: appStackInstance.StackSetId,
+				})
+				require.NoError(t, err)
+
+				err = appCfClient.WaitUntilStackDeleteComplete(&awsCF.DescribeStacksInput{
+					StackName: appStackInstance.StackId,
+				})
+				require.NoError(t, err)
+			}
+			// Delete the StackSet once all the StackInstances are cleaned up
+			_, err = appCfClient.DeleteStackSet(&awsCF.DeleteStackSetInput{
+				StackSetName: aws.String(appStackSetName),
+			})
+			require.NoError(t, err)
+
+			_, err = appCfClient.DeleteStack(&awsCF.DeleteStackInput{
+				StackName: aws.String(appRoleStackName),
+			})
+			require.NoError(t, err)
+			err = appCfClient.WaitUntilStackDeleteComplete(&awsCF.DescribeStacksInput{
+				StackName: aws.String(appRoleStackName),
+			})
+			require.NoError(t, err)
+
+			// delete the environment stack once we are done
+			_, err = envCfClient.DeleteStack(&awsCF.DeleteStackInput{
+				StackName: aws.String(envStackName),
+			})
+			require.NoError(t, err)
+			err = envCfClient.WaitUntilStackDeleteComplete(&awsCF.DescribeStacksInput{
+				StackName: aws.String(envStackName),
+			})
+			require.NoError(t, err)
+		}()
+
+		// Given both the application stack and env we are deploying to do not
+		// exist
+		assertStackDoesNotExist(t, appCfClient, appRoleStackName)
+		assertStackDoesNotExist(t, envCfClient, envStackName)
+
+		// create a stackset
+		err = appDeployer.DeployApp(&deploy.CreateAppInput{
+			Name:      app.Name,
+			AccountID: app.AccountID,
+		})
+		require.NoError(t, err)
+
+		// Deploy the environment in the same tools account but in different
+		// region and wait for it to be complete
+		require.NoError(t, envDeployer.DeployAndRenderEnvironment(os.Stderr, &environmentToDeploy))
+
+		// Ensure that the newly created env stack exists
+		assertStackExists(t, envCfClient, envStackName)
+
+		// Provision resources needed to support a pipeline in a region with
+		// no existing copilot environment.
+		err = appDeployer.AddPipelineResourcesToApp(
+			&app,
+			*appSess.Config.Region)
+		require.NoError(t, err)
+
+		stackInstances, err := appCfClient.ListStackInstances(&awsCF.ListStackInstancesInput{
+			StackSetName: aws.String(appStackSetName),
+		})
+		require.NoError(t, err)
+		require.Equal(t, 1, len(stackInstances.Summaries),
+			"application stack instance should exist")
+
+		resources, err := appDeployer.GetRegionalAppResources(&app)
+		require.NoError(t, err)
+		artifactBuckets := regionalResourcesToArtifactBuckets(t, resources)
+
+		pipelineInput := &deploy.CreatePipelineInput{
+			AppName: app.Name,
+			Name:    pipelineStackName,
+			Source: &deploy.CodeCommitSource{
+				ProviderName:  manifest.CodeCommitProviderName,
+				Branch:        "master",
+				RepositoryURL: "https://us-west-2.console.aws.amazon.com/codesuite/codecommit/repositories/aws-sample/browse",
```

Comment:
Does this repo need to exist in our cicd account?

Reply:
I don't think so, because this test doesn't go as far as CodePipeline sourcing the repo, right? And this passes locally even when I have a dummy repo URL in there. But after we add a repo to the cicd account for e2e tests, we can change this URL if we'd like.

Reply:
Gotchu that make sense, it just creates the stack but the pipeline itself won't succeed 👍